src/App/mermaidData/syncApiDataIntoOfflineStorage/useInitializeSyncApiDataIntoOfflineStorage.js

```diff
@@ -8,47 +8,41 @@ import { useSyncStatus } from './SyncStatusContext'
 
 export const useInitializeSyncApiDataIntoOfflineStorage = ({
   apiBaseUrl,
-  getAccessToken,
   dexiePerUserDataInstance,
   isMounted,
   isAppOnline,
   handleHttpResponseError,
   syncApiDataIntoOfflineStorage,
+  refreshCurrentUser,
 }) => {
   const location = useLocation()
   const isPageReload = useRef(true)
 
   const { setIsSyncInProgress, setIsOfflineStorageHydrated, setSyncErrors } = useSyncStatus()
 
-  const _conditionallySyncOfflineStorageWithApiData = useEffect(() => {
-    const resetSyncErrors = () => {
-      setSyncErrors([])
-    }
-    const appendSyncError = (newError) => {
-      setSyncErrors((previousState) => [...previousState, newError])
-    }
```

Comment:
The linter pointed out that I forgot to add `refreshCurrentUser` to a dep array, and when I did a bunch of tests failed (because it caused the stuff inside a 'useEffect` for syncing to run too much. Most of the changes in this file are addressing that.

Basically I just moved a bunch of vars outside of the original `useEffect` for reuse and created another `useEffect`. One for project list page syncing, another for project-specific page syncing.  In doing this, each useEffect has a more targeted dep array and the innards are more likely to only run when they need to.

Reply:
I like the idea of having a more targeted dep array.  just ran tests on the recent changes however and still seeing failed tests (14 failed)

Reply:
Ugh. That may be an issue with tests more than the feature. @$!$@

Reply:
Also kudos to you for checking out the branch and running it locally :)

Reply:
looks like some failed tests might have creeped into dev ... so yeah might not have to do with this feature!